components/raftstore/src/store/fsm/peer.rs

```diff
@@ -1027,6 +1027,19 @@ where
             self.fsm.peer.mut_store().flush_cache_metrics();
             return;
         }
+
+        if self.fsm.peer.is_leader() && !self.fsm.peer.peers_start_pending_time.is_empty() {
```

Comment:
Isn't it reported to PD already? Can alert manager use the metrics from PD directly?

Reply:
Before this PR, `peers_start_pending_time` is only used for logging.

Reply:
How about updating metrics right before logging? base tick is too frequent for raft tick unrelated work.

Reply:
`peers_start_pending_time` only log when a pending peer has caught up log, so if a pending peer does not catch up log yet it will not show up at the metric, and when it shows up at the metric it already not a pending peer.

To make the metric accurate, we need periodically report the `peers_start_pending_time`, if 1 tick is too frequent maybe 5 or more based tick?

Reply:
The default value of `pd-heartbeat-tick-interval` is `1m`, use metrics from PD is not in time.

Reply:
> if 1 tick is too frequent maybe 5 or more based tick?

Why not put it into a separate tick or put it into pd-heartbeat-tick? Why is it has to be bound to base tick? I have seen people set base-tick-interval to 100ms, what do you think is a suitable tick count?

> use metrics from PD is not in time.

Why `1m` is not in time? What could go wrong if it could only be handled after 1minute?

Reply:
> Why 1m is not in time? What could go wrong if it could only be handled after 1minute?

I thought the `pd-heartbeat-tick` is `10m` at first, `10m` is not in time.

But `pd-heartbeat-tick` is `1m` is acceptable. I will move into pd-heartbeat-tick.